storage/shard.go

```diff
@@ -1446,12 +1463,19 @@ func (s *dbShard) Flush(
 	s.RLock()
 	if s.bs != bootstrapped {
 		s.RUnlock()
+		// TODO: Rename for snapshot
 		return errShardNotBootstrappedToFlush
 	}
 	s.RUnlock()
 
 	var multiErr xerrors.MultiError
-	prepared, err := flush.Prepare(s.namespace, s.ID(), blockStart)
+	prepareOpts := persist.PrepareOptions{
+		NsMetadata: s.namespace,
+		Shard:      s.ID(),
+		BlockStart: blockStart,
+		WrittenAt:  blockStart,
```

Comment:
hm why provide blockStart for writtenAt? should we skip if it's not applicable?

Reply:
Not sure. Setting "zero time" or setting it equal to the blockStart + blockSize seem equally valid to me (I see now that putting blockStart for "writtenAt" is dumb, not sure why I did that to begin with)

Reply:
Any preference @r @prateek ?

Reply:
`WrittenAt` implies something inaccurate here. I think we should rename `WrittenAt` to `SnapshotTime`, and skip it in the case we're not writing snapshot files.

Reply:
done